xonsh/tools.py

```diff
@@ -1686,15 +1686,29 @@ def globpath(s, ignore_case=False, return_empty=False):
 
 def _iglobpath(s, ignore_case=False):
```

Comment:
Yeah, I would also be in favor of a `sorted` kwarg.  This wouldn't be just for testing.  You should then remove the envvar lookup from this function completely, and then look up the envvar at the call site of this function.